Sming/Libraries/DS18S20/ds18s20.h

```diff
@@ -15,7 +15,7 @@
  *  @ingroup    libraries
  *  @{
 */
-#include <Wire.h>
+#include <Libraries/OneWire/OneWire.h>
```

Comment:
I think just a `class OneWire;` declaration would suffice as it's not used in the header, nor by client code.

Reply:
Thanks, @mikee47. I've added `class OneWire;` declaration

Reply:
Just a second. The variable `ds` is used in `ds18s20.cpp`. I still think we should use `OneWire.h` instead of `class OneWire;`

Reply:
Yes, but it already includes the header so it'll be happy :-)

Reply:
Just a question of not introducing a load of additional dependencies into application code.

Reply:
While you're at it, could you also remove the #include from `samples\Temperature_DS1820\app\application.cpp`. It's not needed and shouldn't be there; explains why the error you report hadn't surfaced already.

Reply:
>  remove the #include from `samples\Temperature_DS1820\app\application.cpp`. 

Yep, done.

Reply:
Thanks!